checkov/json/base_registry.py

```diff
@@ -0,0 +1,150 @@
+        skipped_checks,
+        runner_filter,
+        entity,
+        entity_name,
+        entity_type,
+        results,
+    ):
+        for check in checks:
+            skip_info = ([x for x in skipped_checks if x["id"] == check.id] or [{}])[0]
+
+            if runner_filter.should_run_check(check.id, check.bc_id):
+                scanner = self._scanner.get(check.block_type, self._scan_json_document)
+                if check.path:
+                    target = entity
+                    for p in check.path.split("."):
+                        if p.endswith("]"):
+                            ip = p.split("[")
+                            i = int(ip[1][:-1])
+                            target = target[ip[0]][i]
+                        else:
+                            target = target[p]
+                else:
+                    target = entity
+                scanner(
+                    scanned_file,
+                    check,
+                    skip_info,
+                    target,
+                    entity_name,
+                    entity_type,
+                    results,
+                )
+
+    def scan(self, scanned_file, entity, skipped_checks, runner_filter):
+        results = {}
+
+        if not entity:
+            return results
+
+        for instruction, checks in self.checks.items():
+            self._scan_json(
+                scanned_file=scanned_file,
+                checks=checks,
+                skipped_checks=skipped_checks,
+                runner_filter=runner_filter,
+                entity=entity,
+                entity_name=instruction,
+                entity_type=instruction,
+                results=results,
+            )
+
+        if self.wildcard_checks["*"]:
+            self._scan_json(
+                scanned_file=scanned_file,
+                checks=self.wildcard_checks["*"],
+                skipped_checks=skipped_checks,
+                runner_filter=runner_filter,
+                entity=entity,
+                entity_name=scanned_file,
+                entity_type="*",
+                results=results,
+            )
+
+        return results
+
+    def update_result(
+        self,
+        check,
+        entity_configuration,
+        entity_name,
+        entity_type,
+        results,
+        scanned_file,
+        skip_info,
+    ):
+        result = self.run_check(
+            check,
+            entity_configuration,
+            entity_name,
+            entity_type,
+            scanned_file,
+            skip_info,
+        )
+
+        results[check] = {}
+        if result["result"] == CheckResult.SKIPPED:
+            results[check]["result"] = result["result"]
+            results[check]["suppress_comment"] = result["suppress_comment"]
+            results[check]["results_configuration"] = None
+        else:
+            results[check]["result"] = result["result"][0]
+            results[check]["results_configuration"] = result["result"][1]
```

Comment:
@BrentSouza Why do we get the conf back from the check? IF it's the one `update_result` gets, can we leverage that?

Reply:
@nimrodkor I was following along with the Dockerfile implementation, which does something similar. The conf that gets returned back from the check could be a property of the conf that is passed into `update_result`. For example, the check gets invoked for the following json node:

```json
[
  { "prop": true },
  { "prop": false}
]
```
Let's say we have a check that ensures all `prop` values are `true`:

```py
for item in conf:
    if not item["prop"]:
        return CheckResult.FAILED, item
return CheckResult.PASSED  
```

In this case the failing check returns the JSON node that failed, which we can then use to print the relevant code snippet since the node contains the line information as well as the json values.

Does this make sense or would you prefer I follow a different runner implementation?

Reply:
@nimrodkor I thought about this a bit more and in the general case, I think users wouldn't need to return the config, so I made some changes to make returning the failed configuration node optional. Now if you just return the CheckResult, it will use the `entity_configuration` value in `update_result` as you noted.

Another option I can think of to completely get rid of returning a node could be to do something with `get_evaluated_keys` so that it returns the path to the failed key in the configuration. Ideally, this would happen within the base check or base_registry but I haven't fully thought it through yet. What do you think?

Reply:
Love it!